hasktorch/src/Torch/Typed/Functional.hs

```diff
@@ -3899,6 +3912,52 @@ median' input = unsafePerformIO $ ATen.cast3 ATen.Managed.median_tlb
                                         (natValI @dim)
                                         (keepOrDropDimVal @keepOrDropDim)
 
+-- | meanAll
+--
+-- >>> dtype &&& shape $ meanAll (ones :: CPUTensor 'D.Float '[2,2])
+-- (Float,[])
+meanAll
+  :: forall shape dtype device
+   . Tensor device dtype shape -- ^ input
+  -> Tensor device dtype '[] -- ^ output
+meanAll input = unsafePerformIO $ ATen.cast1 ATen.Managed.mean_t input
+
+-- | meanDim
+--
+-- >>> t = ones :: CPUTensor 'D.Float '[3,4,5]
+-- >>> dtype &&& shape $ fst $ meanDim @0 t
+-- (Float,[4,5])
+-- >>> dtype &&& shape $ fst $ meanDim @1 t
+-- (Float,[3,5])
+-- >>> dtype &&& shape $ fst $ meanDim @2 t
+-- (Float,[3,4])
+meanDim
+  :: forall d shape dtype device
+   . (KnownNat d)
+  => Tensor device dtype shape -- ^ input
+  -> ( Tensor device dtype    (DropValue shape d)
+     , Tensor device 'D.Int64 (DropValue shape d)
+     ) -- ^ output
+meanDim input = unsafePerformIO $ ATen.cast2 ATen.Managed.mean_tl input (natValI @d)
```

Comment:
https://hasktorch.github.io/libtorch-ffi-1.4.0.0/html/Torch-Internal-Managed-Native.html#v:mean_tl
mean_tl returns not tuple but a Tensor.